aiohttp/helpers.py

```diff
@@ -741,3 +742,63 @@ def set_result(fut, result):
 def set_exception(fut, exc):
     if not fut.done():
         fut.set_exception(exc)
+
+
+class ChainedProps(Mapping):
```

Comment:
I think the ChainedMaps would be better name here since it's a mapping. Props suffix leaves the feeling that it's something about attr-based access. While ChainedMaps causes collision with stdlib ChainMap, still, they have a bit different naming what should cause suspicions. However, if we make this class for internal-only usage, than own ChainMap is fine.

Reply:
Naming is the hardest thing!
Maybe we can invite a name without *chain* word?

Reply:
Hm...How about to just a Mapping? With the trick that all the maps will get merged right at the moment of it creation?

Reply:
I mean, if we do merge all the maps, than this class has no reason to exists. The merge result could be returned as MappingProxy - well known immutable mapping.

Reply:
Let's name it `ChainMapProxy` and keep existing implementation -- I don't like early merging.